### ISSUE TITLE
Uncaught ReferenceError: regeneratorRuntime is not defined

### ISSUE THREAD

**Issue by jonasdumas** [kettanaito/react-advanced-form#119]
Hello,

It's 3 days i'm spending time to find the right solution for my app's form.
I just found your react-advanced-form library and like the concept which seems to fit exactly what i'm looking for.

Unfortunately, I can't make it work. I have this error in console:

```
Uncaught ReferenceError: regeneratorRuntime is not defined
    at index.js:1
    at Object.<anonymous> (index.js:1)
    at __webpack_require__ (index.js:1)
    at Object.<anonymous> (index.js:1)
    at __webpack_require__ (index.js:1)
    at Object.<anonymous> (index.js:1)
    at __webpack_require__ (index.js:1)
    at Object.<anonymous> (index.js:1)
    at __webpack_require__ (index.js:1)
    at Object.<anonymous> (index.js:1)
    at __webpack_require__ (index.js:1)
    at index.js:1
    at index.js:1
    at ./node_modules/react-advanced-form/lib/index.js (index.js:1)
    at Object../node_modules/react-advanced-form/lib/index.js (index.js:1)

```
I've required dependencies installed:
**react @ 16.2.0
immutable @ 3.8.2**

Anyone have an idea ?

**Comment by jonasdumas** [comment]
Update: 
I've imported this before:
`import "babel-polyfill";`

And the error gone but then i've this error onblur on any empty field:
```
Uncaught (in promise) TypeError: Cannot read property 'getIn' of undefined
    at Object.t.default [as getErrorMessage] (http://localhost:3000/static/js/bundle.js:56291:24787)
    at Form.<anonymous> (http://localhost:3000/static/js/bundle.js:56291:31621)
    at tryCatch (http://localhost:3000/static/js/bundle.js:81786:40)
    at Generator.invoke [as _invoke] (http://localhost:3000/static/js/bundle.js:82024:22)
    at Generator.prototype.(anonymous function) [as next] (http://localhost:3000/static/js/bundle.js:81838:21)
    at step (http://localhost:3000/static/js/bundle.js:56291:25672)
    at http://localhost:3000/static/js/bundle.js:56291:25772
    at <anonymous>
```

Here's the code I'm testing:
```
import React from 'react';
import "babel-polyfill";
import {Form, Field} from 'react-advanced-form';

export default class MyForm extends React.Component {
  handleSubmit = ({serialized}) => {
    //
  }

  render() {
    return (<Form action={this.handleSubmit}>
      <Field.Input name="username" required="required"/>
      <Field.Input name="password" type="password" required="required"/>
      <Field.Radio name="gender" value="male"/>
      <Field.Radio name="gender" value="female"/>
      <button type="submit">Submit</button>
    </Form>);
  }
}

```

**Comment by kettanaito** [comment]
@jonasdumas hi.

Indeed, regenerator runtime error was related to babel-polyfill. Thanks for pointing out, I will try to include regenerator polyfill within the lib or at least write about babel polyfill needed.


Regarding the last error message, please provide "rules" and "messages" prop to the Form component:
<Form rules={{}} messages={{}}>...</Form>

Please read more on how to declare those validation rules and messages in the official documentation.

Sorry, I don't have an access to my laptop right now to assist you more. Passing rules will solve your error. I need to consider
A. Not demanding rules and messages by default
B. At least provide better warning messages

**Comment by kettanaito** [comment]
Here are the docs on how to declare rules and messages:
https://kettanaito.gitbooks.io/react-advanced-form/validation/validation-rules.html

https://kettanaito.gitbooks.io/react-advanced-form/validation/validation-messages.html

Please let me know that helped.

**Comment by kettanaito** [comment]
I've opened an issue to make rules and messages props optional: https://github.com/kettanaito/react-advanced-form/issues/120.

I will cover this after the holidays.

**Comment by kettanaito** [comment]
Opened a dedicated issue to handle this regenerator runtime situation: https://github.com/kettanaito/react-advanced-form/issues/122.

**Comment by jonasdumas** [comment]
Hello @kettanaito 
Thank you for your answers. 
I spents few hours today to study and implement your solution. I could make it work easily thanks to your remarks. You did an amazing work. It's exactly what I was expected from a validation form system. 
I've tested a lot before and your one is the number one for many reasons!

I got 3 other questions/requests but I'll open new issues to make research easier for futur users.
